### ISSUE TITLE
Missing Email Ping Listener

### ISSUE THREAD

**Issue by alexsidow** [healthchecks/healthchecks#139]
I was looking to setup healthchecks on premise with using UUID@hostname but there would be a SMTP listener/daemon required I can't find in healthchecks. How is this supposed to be set up or start up? This part seems to be missing in the repo whilst working properly on hchk.io

Thank you.

**Comment by cuu508** [comment]
Hi @alexsidow, you are correct, this is currently missing from the python codebase. hchk.io runs custom Go code which I'm not quite ready to publish.

I'll see if I can put together a listener based on python's [smtpd](https://docs.python.org/3/library/smtpd.html) module. It would likely be a management command--

    $ python manage.py smtpd 25
    Listening for emails on port 25

I have not used the smtpd module before so not yet sure if this can work or not.

**Comment by iphoting** [comment]
Alternatively, if Mailgun mail provider is used (typically so, if you're hosting on Heroku), the [Mailgun incoming mail API][1] could be used to convert incoming email pings into HTTP REST hits.

[1]: https://documentation.mailgun.com/en/latest/quickstart-receiving.html#how-to-start-receiving-inbound-email

**Comment by cuu508** [comment]
Good point. I've also used AWS SES for handling inbound emails. In SES you can specify that the inbound emails should be passed to an AWS Lambda function. The function I used for reference:

```
var https = require('https');

exports.handler = function(event, context) {
    var sesNotification = event.Records[0].ses;
    console.log("SES Notification:", sesNotification);

    var recipient = sesNotification.mail.destination[0];
    var parts = recipient.split("@");

    var options = {
        host: 'hchk.io',
        port: 443,
        path: '/' + encodeURIComponent(parts[0]),
        method: 'GET',
        headers: {
            'User-Agent': 'Email from ' + sesNotification.mail.source
        }
    };

    var req = https.request(options, function(res) {
        console.log(res.statusCode);
        context.succeed();
    });
    req.end();

    req.on('error', function(e) {
        context.fail("Call to hchk.io fails: ", e);
    });
};
```

**Comment by cuu508** [comment]
I've just added a management command "smtpd" which listens for emails on the specified ip and port. Invoke it like this:

    $ ./manage.py smtpd --host=1.2.3.4 --port=2525

The default value for host is 0.0.0.0 (all interfaces), and the default port is 25. 
25 is a privileged port, and I'm not sure what's the best way for a python program to bind to it. Doing `setcap 'cap_net_bind_service=+ep' /usr/bin/python3.5` doesn't seem like a good idea... I haven't tested it, but looks like port 2525 might also work.

So far I have only tested this command locally, and not with real email clients like Gmail.
